sympy/ntheory/primetest.py

```diff
@@ -6,51 +6,64 @@
 from itertools import count
 
 from sympy.core.sympify import sympify
-from sympy.external.gmpy import gmpy as _gmpy, jacobi, is_square as gmpy_is_square, bit_scan1
+from sympy.external.gmpy import (gmpy as _gmpy, gcd, jacobi,
+                                 is_square as gmpy_is_square,
+                                 bit_scan1)
 from sympy.utilities.misc import as_int
 
 
-def is_euler_pseudoprime(n, b):
-    """Returns True if n is prime or an Euler pseudoprime to base b, else False.
+def is_euler_pseudoprime(n, a):
+    r"""Returns True if ``n`` is prime or is an odd composite integer that
+    is coprime to ``a`` and satisfy the modular arithmetic congruence relation:
 
-    Euler Pseudoprime : In arithmetic, an odd composite integer n is called an
-    euler pseudoprime to base a, if a and n are coprime and satisfy the modular
-    arithmetic congruence relation :
-
-    a ^ (n-1)/2 = + 1(mod n) or
-    a ^ (n-1)/2 = - 1(mod n)
+    .. math ::
+        a^{(n-1)/2} \equiv \pm 1 \pmod{n}
 
     (where mod refers to the modulo operation).
 
+    Parameters
+    ==========
+
+    n : Integer
+        ``n`` is a positive integer.
+    a : Integer
+        ``a`` is a positive integer.
+        ``a`` and ``n`` should be relatively prime.
+
+    Returns
+    =======
+
+    bool : If ``n`` is prime, it always returns ``True``.
+           The composite number that returns ``True`` is called an Euler pseudoprime.
+
     Examples
     ========
 
     >>> from sympy.ntheory.primetest import is_euler_pseudoprime
-    >>> is_euler_pseudoprime(2, 5)
-    True
+    >>> from sympy.ntheory.factor_ import isprime
+    >>> for n in range(1, 1000):
+    ...     if is_euler_pseudoprime(n, 2) and not isprime(n):
+    ...         print(n)
+    341
+    561
 
     References
     ==========
 
     .. [1] https://en.wikipedia.org/wiki/Euler_pseudoprime
     """
-
-    if not mr(n, [b]):
+    n, a = as_int(n), as_int(a)
+    if a < 1:
+        raise ValueError("a should be an integer greater than 0")
+    if n < 1:
+        raise ValueError("n should be an integer greater than 0")
+    if n == 1:
         return False
```

Comment:
```suggestion
        return False
    if a == 1:
        return bool(n % 2)  # only True for odd (prime or composite)
```

Reply:
In keeping with docstring, we don't want to return True for all n when `a==1`; modification suggested.

Reply:
Even with the current RP, when `a=1`, the result is as expected. Conversely, modifying it that way would result in `False` when `n=2, a=1`.

Reply:
I updated the suggestion so True is returned for any prime (all odd except for 2) or odd composite. The docstring indicates that we should not be returning True for even numbers that are not prime (which the currrent code does, I believe).

Reply:
To clarify and avoid any misunderstandings, the requirement for n to be odd is to calculate $(n-1)/2$. Therefore, it is necessary to handle even numbers separately before calculating $(n-1)/2$. It is rational to consider 2 as a prime number and classify all other even numbers as not prime.